File: solutionFiles/RomantoInteger.py
import json
import sys

def roman_to_int(s: str) -> int:
    roman_to_value = {
        'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000
    }
    total = 0
    prev_value = 0

    for char in reversed(s):
        value = roman_to_value[char]
        if value < prev_value:
            total -= value
        else:
            total += value
        prev_value = value

    return total

def main():
    if len(sys.argv) != 2:
        print("Usage: python script.py '<json_string>'")
        return

    input_json = sys.argv[1]
    try:
        data = json.loads(input_json)
        roman_numeral = data.get("roman")
        if not roman_numeral:
            print(json.dumps({"error": "No roman numeral found"}))
            return

        integer_value = roman_to_int(roman_numeral)
        result = json.dumps({"result": integer_value})
        print(result)
    except json.JSONDecodeError:
        print(json.dumps({"error": "Invalid JSON input"}))

if __name__ == "__main__":
    main()

File: solutionFiles/RomantoInteger.js
const process = require('process');

function romanToInt(s) {
    const romanToValue = {
        'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000
    };
    let total = 0;
    let prevValue = 0;

    for (let i = s.length - 1; i >= 0; i--) {
        const value = romanToValue[s[i]];
        if (value < prevValue) {
            total -= value;
        } else {
            total += value;
        }
        prevValue = value;
    }

    return total;
}

function main() {
    if (process.argv.length !== 3) {
        console.log("Usage: node script.js '<json_string>'");
        return;
    }

    const inputJson = process.argv[2];
    try {
        const data = JSON.parse(inputJson);
        const romanNumeral = data.roman;
        if (!romanNumeral) {
            console.log(JSON.stringify({ error: "No roman numeral found" }));
            return;
        }

        const integerValue = romanToInt(romanNumeral);
        const result = JSON.stringify({ result: integerValue });
        console.log(result);
    } catch (e) {
        console.log(JSON.stringify({ error: "Invalid JSON input" }));
    }
}

main();
